bisect.c

```diff
@@ -969,11 +975,12 @@ int bisect_next_all(const char *prefix, int no_checkout)
 
 	if (!revs.commits) {
 		/*
-		 * We should exit here only if the "bad"
+		 * We should return error here only if the "bad"
 		 * commit is also a "skip" commit.
 		 */
-		exit_if_skipped_commits(tried, NULL);
-
+		res = error_if_skipped_commits(tried, NULL);
+		if (res)
+			return res;
```

Comment:
The function is still exit()ing below, so I wonder if it's the right step to start making this function return errors.

If this commit is just about changing `exit_if_skipped_commits()` into `error_if_skipped()`, it might want to try to not change the behavior of other functions.

Changing the behavior of `bisect_next_all()` might then be done in a separate commit.

Reply:
I don't think it changes the behavior of `bisect_next_all()`. It only handles the `res` declaration for `error_is_skipped_commits()` function call. And as libify `exit_if_skipped_commits()` into `error_if_skipped()` needs changes in `bisect_next_all()`, so it should be part of the same commit.
Correct me if I am wrong.

Reply:
I agree that changing `exit_if_skipped_commits()` into `error_is_skipped_commits()` needs changes in `bisect_next_all()`. What I am saying is that the commit would be easier to review if the global behavior of `bisect_next_all()` and the function it calls would stay the same. So that reviewers don't need to also think about what happens where `bisect_next_all()` is called.

For example if the above change was like:

```
  		res = error_if_skipped_commits(tried, NULL);
		if (res)
			exit(-res);
```

it would be clear that the global behavior of `bisect_next_all()` isn't changed by this commit.

Reply:
Okay, Should I implement this change in libify `bisect_next_all()` commit or in a separate commit altogether?

Reply:
I think that overall it would be better to have one commit that only changes `exit_if_skipped_commits()` into `error_is_skipped_commits()` and one separate commit that only libifies `bisect_next_all()`.

You already have https://github.com/gitgitgadget/git/pull/117/commits/88156b20e3436f8ef18029aca466add2fe078eaf that libifies `bisect_next_all()`, so you could move there the changes that are not related to changing `exit_if_skipped_commits()` into `error_is_skipped_commits()`.